reconcile/terraform_resources.py

```diff
@@ -717,16 +745,14 @@ def run(
     # populate the resource inventory with latest output data
     populate_desired_state(ri, resource_specs)
 
-    actions = ob.realize_data(
-        dry_run, oc_map, ri, thread_pool_size, caller=account_name
-    )
+    actions = ob.realize_data(dry_run, oc_map, ri, thread_pool_size, caller=acc_name)
 
     if not light and tf.should_apply:
         disable_keys(
             dry_run,
             thread_pool_size,
             disable_service_account_keys=True,
-            account_name=account_name,
+            account_name=acc_name,
```

Comment:
account_name acts as a filter in disable_keys. by calling it this way, disable_key will only act on the first passed account

Reply:
The idea is that we never get here with more than one account.

Reply:
Why is that?

Reply:
the code does not prevent that though and there might be the impression that it is a supported run model.
you could put some guarding code in place that fails the integration if it reaches this part of the code with more than one account_name

Reply:
There is even a test for that
https://github.com/app-sre/qontract-reconcile/pull/3042/files#diff-aa3407e2884577fb05f6ed1925376a267ff8655228672c67c899a3bafe4c7497R6-R29